shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -187,12 +202,50 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
     // engine since it sent this update, and needs to now be made to match the
     // engine's version of the state.
     [self updateEditState];
+  } else if ([method isEqualToString:@"TextInput.setEditableSizeAndTransform"]) {
+    NSDictionary* state = call.arguments;
+    [self setEditableTransform:state[@"transform"]];
+  } else if ([method isEqualToString:@"TextInput.setCaretRect"]) {
```

Comment:
Added comments on https://github.com/flutter/flutter/pull/77608. I'd have thought that `TextInput.setMarkedTextRect` should be sufficient here -- despite its name, it should emit the current caret rect when the composing range is empty.

Reply:
As noted in https://github.com/flutter/flutter/pull/77608, `TextInput.setMarkedTextRect` only works while compositing. So we need another method for caret position.